app/Offer/OfferControllerProvider.php

```diff
@@ -105,23 +101,18 @@ function (Application $app) {
                     return new OfferPermissionsController(
                         $permissionsToCheck,
                         $app['offer_permission_voter'],
-                        $currentUserId
+                        $app['current_user_id'] ? new StringLiteral($app['current_user_id']) : null
```

Comment:
Should we try to avoid using a `StringLiteral` here instead of a plain `string`?

Reply:
This controller passes the user id down to the permission voter injected above in `isAllowed` calls. So it would mean we'd have to refactor the permission voter as well to not use `StringLiteral`. Or we need to inject it as a nullable string and then wrap it in StringLiteral before passing it to the permission voter, but then we need to do another null check when wrapping it. So I did it here so I can just do it in this null check already.